lib/websocket_proxy.rb

```diff
@@ -49,8 +49,8 @@ def transmit(sockets, is_ws)
 
   def cleanup
     @console.destroy unless @console.destroyed?
-    @sock.close unless @sock.closed?
-    @ws.close unless @ws.closed?
+    @sock.close unless @sock.nil? || @sock.closed?
+    @ws.close unless @ws.nil? || @ws.closed?
```

Comment:
I'm not a fan of `unless` with an `||`.  While this one is easy to follow, `if` is easier with a compound conditional since it's not inverted.

```ruby
    @sock.close if @sock && !@sock.closed?
    @ws.close if @ws && !@ws.closed?
```